In [6]:
import numpy as np
from scipy import stats
import numpy as np

### Задача 1

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

### Решение:

#### 1. Формулируются нулевая и альтернативная гипотезы.
1) Гипотеза $H_0$: средний диаметр шарика для подшипников равен 17 мм.

2) Гипотеза $H_1$: средний диаметр шарика для подшипников больше 17 мм.

#### 2. Задаётся некоторая функция Т от выборки
Предполагается, что количество шариков большое, то распределение диаметра шариков является нормальным или близким к нему. Т.к. случайная величина диаметра имеет нормальное распределение, то среднее значение повыборке $T$ будет также иметь нормальное распределение с параметрами $a = 17$ и $\sigma = \sqrt{\dfrac{D(X)}{n}} = \sqrt {\dfrac{4}{100}} = 0.2 $

In [7]:
a = 17
m = 17.5
n = 100
D = 4
sigma = np.sqrt(D/n)
sigma

0.2

#### 3. Фиксируется уровень значимости  𝛼
- допустимая для данной задачи вероятность ошибки первого рода (чаще всего 0.01, 0.05 или 0.1).

In [8]:
alpha = 0.05

#### 4. Определяется критическая область $\Omega_\alpha = (t_{1 - \alpha}, \infty)$.

Выбранный уровень значимости $\alpha = 0.05$ означает, что для проверки гипотезы о среднем значении a = 17 требуется, чтобы с вероятностью 95% выборочное среднее попадало в правосторонний интервал, который (по правилу 2-х сигм) равен $(a, a + 2 \cdot \sigma)$.

То есть критическая область лежит правее точки $a + 2 \cdot \sigma $ и если выборочное среднее = 17.5 попадёт в критическую область, то основная гипотеза отвергается.

In [9]:
(a, a + 2*sigma)

(17, 17.4)

#### 5. Проводится статистический тест
Для конкретной выборки  𝑋  считается значение  𝑇(𝑋) , и если оно принадлежит  Ω𝑎𝑙𝑝ℎ𝑎 , то заключаем, что данные противоречат гипотезе  𝐻0 , и принимается гипотеза  𝐻1.

Гипотеза о том, что шарики для подшипников имеют средний диаметр 17 мм, при выбранном уровне значимости $\alpha = 0.05$ отвергается.

### Задача 2

Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочное среднее равно 80, а объём равен 256.

### Решение:

Найдем доверительный интервал согласно формуле: $$
P \left( 
\overline{X} - t_{1 - \alpha/2, \: n - 1} \cdot \dfrac{S}{\sqrt{n}} 
\leq M(X) \leq
\overline{X} + t_{1 - \alpha/2, \: n - 1} \cdot \dfrac{S}{\sqrt{n}} 
\right) = 1 - \alpha.
$$

где
$\overline{X} -  = 80$,
$S = 16$,
$n = 256$,

In [13]:
std_ = 16
mean_ = 80
n = 256
print ('Доверительный интервал равен:', (mean_ - 2*std_/np.sqrt(n), mean_ + 2*std_/np.sqrt(n)))

Доверительный интервал равен: (78.0, 82.0)


### Задача 3

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:

202, 203, 199, 197, 195, 201, 200, 204, 194, 190

Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

### Решение:
Найдем доверительный интервал, чтобы проверить гипотезу, что среднее значение веса пачки 200 г.

In [20]:
sample = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
m = sample.mean()
n = 10
sigma = np.sqrt(((sample - m)**2).sum()/(n-1))
print ('𝜎 =', sigma)

𝜎 = 4.453463071962462


Табличное значение коэффициента Стьюдента

In [21]:
t_int = 2.8214

Проверим, что коэффицент Стьюдента = 2.8214, соответствует вероятности =

In [22]:
t = stats.t(n-1)
t.cdf(2.8214)

0.9899993808040004

In [23]:
print ('Доверительный интервал равен:', (m - t_int*sigma/np.sqrt(n), m + t_int*sigma/np.sqrt(n)))

Доверительный интервал равен: (194.5265978950862, 202.4734021049138)


Доверительный интервал, построенный с учётом доверительной вероятности 99%, = (194.5, 202.4), включает значение предполагаемого среднего значения а = 200 => утверждение продавца, что средний вес пачки = 200 гр, верно.